setup/build.py

```diff
@@ -162,20 +166,18 @@ def init_env():
     if islinux:
         cflags.append('-pthread')
         ldflags.append('-shared')
-        cflags.append('-I'+sysconfig.get_python_inc())
-        ldflags.append('-lpython'+sysconfig.get_python_version())
 
     if isbsd:
         cflags.append('-pthread')
         ldflags.append('-shared')
-        cflags.append('-I'+sysconfig.get_python_inc())
-        ldflags.append('-lpython'+sysconfig.get_python_version())
 
     if ishaiku:
         cflags.append('-lpthread')
         ldflags.append('-shared')
+
+    if islinux or isbsd or ishaiku:
         cflags.append('-I'+sysconfig.get_python_inc())
-        ldflags.append('-lpython'+sysconfig.get_python_version())
+        ldflags.extend(shlex.split(sysconfig.get_config_var('BLDLIBRARY')))
```

Comment:
At least on my system, BLDLIBRARY insets both -L and -l flags. That could have unexpected side effects. A more direct replacement would be:

```py
'-lpython' + sysconfig.get_config_var('VERSION') + getattr(sys, 'abiflags', '')

Reply:
I got `-L.` on most systems I tested this function on too, although it didn't seem to harm anything at this time.

Why `getattr(sys, 'abiflags', '')` instead of `sys.abiflags`?